systemd/systemd.go

```diff
@@ -162,13 +162,15 @@ func (*systemd) DaemonReload() error {
 func (s *systemd) Enable(serviceName string) error {
 	enableSymlink := filepath.Join(s.rootDir, snapServicesDir, servicesSystemdTarget+".wants", serviceName)
 
-	serviceFilename := filepath.Join(s.rootDir, snapServicesDir, serviceName)
 	// already enabled
 	if _, err := os.Lstat(enableSymlink); err == nil {
 		return nil
 	}
 
-	return os.Symlink(serviceFilename[len(s.rootDir):], enableSymlink)
+	// important, do nt use the s.rootDir here as this is the
+	// real name (oldname)
```

Comment:
There's a small typo there. I suggest something like this, assuming it is what is meant:

    // Do not use the s.rootDir here. The link must point to the real path.

Out of curiosity, what is _(oldname)_ referring to?